### PR TITLE
Support setting Nessus web server port

### LINKED ISSUE #52: Make Nessus listen on port 443
Modify the Nessus installation so that it listens on port 443, rather than the default of 8834.  Even better would be to make the port a Terraform variable with a default of 8834.

See [this page](https://docs.tenable.com/nessus/commandlinereference/Content/ManageAdvancedSettings.htm) to see how to easily change the `xmlrpc_listen_port` via a `nessuscli` command.

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds a new optional Terraform variable `nessus_web_server_port`, which when set will modify the web server port used by any Nessus instances that are deployed.  The port defaults to 8834, which is the default port used by Nessus.

## 💭 Motivation and context ##

If the Nessus port is set to 443, it will make life simpler for assessors who connect to the Nessus web server using a browser.  This also gives us more flexibility in general if the port ever needs to be set to something specific.

Resolves #52.

## 🧪 Testing ##

I deployed a fresh Nessus instance using the code changes in this PR, set `nessus_web_server_port` to 443, and verified that the script ran as expected and that the port was indeed modified.

Log snippet (`/var/log/cloud-init-output.log`):
```console
Setting Nessus web server port to 443...
Successfully set 'xmlrpc_listen_port' to '443'.
The Nessus web server will be restarted.
```

Manual verification that port was correctly changed to 443:
```console
$ sudo /opt/nessus/sbin/nessuscli fix --show xmlrpc_listen_port

--------------------------------------------------
  User Interface
--------------------------------------------------
  * Nessus Web Server Port (xmlrpc_listen_port)
    - Current value: 443
    - Description: Port for the Nessus web server to listen on.
```

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Coordinate with Operations and Assessment teams to make them aware of this new feature.

### CODE REVIEW

File: cloud-init/nessus-setup.sh
Comment (jmorrowomni) [inline cloud-init/nessus-setup.sh]:
```suggestion
echo "Setting Nessus web server port to ${nessus_web_server_port}..."
```

**Review by jmorrowomni — CHANGES_REQUESTED** [review]
Adding "Nessus" to the output will clarify which web server port is current being set and be consistent with other output

**Review by jmorrowomni — APPROVED** [review]

File: variables.tf
Comment (jsf9k) [inline variables.tf]: We _could_ verify that the number is within the expected range (`[1, 65535]`).  This is just a thought and need not impede the approval of this PR.

**Review by jsf9k — APPROVED** [review]
I have one suggestion, which you can feel free to ignore.

**Review by mcdonnnj — APPROVED** [review]

File: variables.tf
Comment (dav3r) [reply variables.tf]: Excellent idea - please see https://github.com/cisagov/cool-assessment-terraform/pull/222/commits/8e332594c7e1dfa9f9df6211da16586f4dd345c3.  It is a little wonky because I don't think Terraform has a built-in function to check that a number is an integer, so I check to make sure that the string representation of the number doesn't contain a period.  I tested it out and it works as intended.

Comment (dav3r) [reply variables.tf]: Arg- looks like `strcontains()` wasn't added until Terraform 1.5, so it is [failing in our current linter](https://github.com/cisagov/cool-assessment-terraform/actions/runs/5871426698/job/15920738947?pr=222#step:15:91) because we are still on [Terraform 1.2.3](https://github.com/cisagov/setup-env-github-action/blob/e40f45df94a2084271806c3651c7ffe177a9d9ef/action.yml#L43-L44).

I will fix this, but probably not until tomorrow.

Comment (jsf9k) [reply variables.tf]: Is there any reason not to update [cisagov/setup-env-github-action](https://github.com/cisagov/setup-env-github-action)?  I'm rockin' 1.5.4 locally over here.

Comment (dav3r) [reply variables.tf]: Not that I'm aware of.  I am going to make a PR shortly in that repo.

Comment (dav3r) [reply variables.tf]: https://github.com/cisagov/setup-env-github-action/pull/66 has been created to address this.  I will resolve this conversation after that PR has been merged and I verify that this branch passes linting.

**Review by jsf9k — APPROVED** [review]
_I love it!_

File: variables.tf
Comment (dav3r) [reply variables.tf]: https://github.com/cisagov/setup-env-github-action/pull/66 has been merged and the tests in this PR are now passing. @jsf9k are you good with resolving this conversation?

Comment (jsf9k) [reply variables.tf]: Yep.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Please update https://github.com/cisagov/cool-assessment-terraform/blob/78996187fde58028b5debae30a207f3c683b230d/versions.tf#L3 to reflect the version requirement mentioned in https://github.com/cisagov/cool-assessment-terraform/pull/222#discussion_r1295084866 to ensure an expected experience.

**Comment by dav3r** [comment]
> Please update
> 
> https://github.com/cisagov/cool-assessment-terraform/blob/78996187fde58028b5debae30a207f3c683b230d/versions.tf#L3
> 
> to reflect the version requirement mentioned in [#222 (comment)](https://github.com/cisagov/cool-assessment-terraform/pull/222#discussion_r1295084866) to ensure an expected experience.

Excellent catch!  Remedied in https://github.com/cisagov/cool-assessment-terraform/pull/222/commits/94c796e31ad3d3135e2d3481c761751ca83bfb8b.

**Review by mcdonnnj — APPROVED** [review]
_Now better than ever._

**Pull request merged**
